File: src/01-easy/lumen/solution.js
const n=+readline(),l=+readline();
const g=[];for(let i=0;i<n;i++) g.push(readline().split(' '));
let dark=0;
for(let r=0;r<n;r++) for(let c=0;c<n;c++){
    let lit=false;
    for(let r2=0;r2<n&&!lit;r2++) for(let c2=0;c2<n&&!lit;c2++)
        if(g[r2][c2]==='C'&&Math.max(Math.abs(r-r2),Math.abs(c-c2))<l) lit=true;
    if(!lit) dark++;
}
console.log(dark);

File: src/01-easy/lumen/solution.kt
import kotlin.math.abs
import kotlin.math.max
fun main(){
    val n=readLine()!!.trim().toInt();val l=readLine()!!.trim().toInt()
    val g=Array(n){readLine()!!.trim().split(" ")}
    var d=0
    for(r in 0 until n) for(c in 0 until n){
        val lit=(0 until n).any{r2->(0 until n).any{c2->g[r2][c2]=="C"&&max(abs(r-r2),abs(c-c2))<l}}
        if(!lit) d++
    }
    println(d)
}
